type_description_interfaces/srv/GetTypeDescription.srv

```diff
@@ -0,0 +1,19 @@
+string type_name
+string type_version_hash
+---
+# True if the type description information is available and populated in the response
+bool successful
+# Empty if 'successful' was true, otherwise contains details on why it failed
+string failure_reason
+
+# The idl or msg file name
+string type_description_raw_file_name
+# The idl or msg file, with comments and whitespace
+# The file extension and/or the contents can be used to determine the format
+string type_description_raw
```

Comment:
I think this was brought up in the REP but maybe never resolved, but this should contain the raw descriptions recursively and therefore we need to have a format for storing them.

We should probably figure that out and document it here. Or tentatively we could have a TODO for that and fix it up later, because presumably we can just update the comment without changing the actual interface of this service.

Reply:
I've added a comment with my proposal for the format. It follows a ROS-classic precedent that I've seen somewhere, which is currently used in MCAP schema records. That is, use a delimiter line of 80 * '=' between individual file contents:

```
file_contents
================================================================================
```

Related, do you think we should we make the raw output optional, specified by a `bool` in the request?

Reply:
I think that works.

As for the option, we discussed that in the REP and decided against it, but we can look at that again if we think we need it.